### ISSUE TITLE
Failed to build shared library on macOS

### ISSUE THREAD

**Issue by mujx** [matrix-org/olm#10]
Running `make` produces the following
```
clang++ -Wall -Werror -g -O3 --shared -fPIC \
            -Wl,-soname,libolm.so.2 \
            -Wl,--version-script,version_script.ver \
            -o build/libolm.so.2.2.2 build/release/src/account.o build/release/src/base64.o build/release/src/cipher.o build/release/src/crypto.o build/release/src/memory.o build/release/src/message.o build/release/src/olm.o build/release/src/pickle.o build/release/src/ratchet.o build/release/src/session.o build/release/src/utility.o build/release/src/ed25519.o build/release/src/error.o build/release/src/inbound_group_session.o build/release/src/megolm.o build/release/src/outbound_group_session.o build/release/src/pickle_encoding.o build/release/lib/crypto-algorithms/sha256.o build/release/lib/crypto-algorithms/aes.o build/release/lib/curve25519-donna/curve25519-donna.o
ld: unknown option: -soname
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [build/libolm.so.2.2.2] Error 1
```
`-soname` and `*.so` are invalid on macOS.

Also the `install` target doesn't work. It requires a shared library although a static build could also work.

**Comment by theilgaard** [comment]
I have the same issue on macOS High Sierra.

```
$ c++ --version
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by theilgaard** [comment]
I can get the library to build by replace `-soname` with `-install_name` and completely removing the lines with `-version-script`

**Comment by mujx** [comment]
The solution for this should be to switch to cmake and let it take care of things like that on each platform.

**Comment by theilgaard** [comment]
Probably, but I am in no way a cmake expert, so I will leave that pull request to other contenders :)

**Comment by uhoreg** [comment]
This should be fixed by the cmake support in 3.0.0
